README.md

```diff
@@ -1,97 +1,46 @@
+
+## Examples
+
+### List packages in the working directory excluding vendor/*
+
+```yml
+---
+format_version: '8'
+default_step_lib_source: https://github.com/bitrise-io/bitrise-steplib.git
+project_type: other
+workflows:
+  release:
+    steps:
+    - git-clone: {}
+    - go-list@0:
+        inputs:
+        - exclude: vendor/*
 ```
 
-## How to create your own step
-
-1. Create a new git repository for your step (**don't fork** the _step template_, create a _new_ repository)
-1. Copy the [step template](https://github.com/bitrise-steplib/step-template) files into your repository
-1. Fill the `step.sh` with your functionality
-1. Wire out your inputs to `step.yml` (`inputs` section)
-1. Fill out the other parts of the `step.yml` too
-1. Provide test values for the inputs in the `bitrise.yml`
-1. Run your step with `bitrise run test` - if it works, you're ready
-
-**For Step development guidelines & best practices** check this documentation: [https://github.com/bitrise-io/bitrise/blob/master/\_docs/step-development-guideline.md](https://github.com/bitrise-io/bitrise/blob/master/_docs/step-development-guideline.md).
-
-**NOTE:**
-
-If you want to use your step in your project's `bitrise.yml`:
-
-1. git push the step into it's repository
-2. reference it in your `bitrise.yml` with the `git::PUBLIC-GIT-CLONE-URL@BRANCH` step reference style:
-
-```
-- git::https://github.com/user/my-step.git@branch:
-   title: My step
-   inputs:
-   - my_input_1: "my value 1"
-   - my_input_2: "my value 2"
-```
+## Configuration
 
-You can find more examples of step reference styles
-in the [bitrise CLI repository](https://github.com/bitrise-io/bitrise/blob/master/_examples/tutorials/steps-and-workflows/bitrise.yml#L65).
+### Inputs
 
-## How to contribute to this Step
+| Parameter | Description | Required | Default |
+| --- | --- | --- | --- |
+| exclude | Exclude patterns | - | "*/vendor/*" |
 
-1. Fork this repository
-1. `git clone` it
-1. Create a branch you'll work on
-1. To use/test the step just follow the **How to use this Step** section
-1. Do the changes you want to
-1. Run/test the step before sending your contribution
+### Outputs
 
-- You can also test the step in your `bitrise` project, either on your Mac or on [bitrise.io](https://www.bitrise.io)
-- You just have to replace the step ID in your project's `bitrise.yml` with either a relative path, or with a git URL format
-- (relative) path format: instead of `- original-step-id:` use `- path::./relative/path/of/script/on/your/Mac:`
-- direct git URL format: instead of `- original-step-id:` use `- git::https://github.com/user/step.git@branch:`
-- You can find more example of alternative step referencing at: https://github.com/bitrise-io/bitrise/blob/master/_examples/tutorials/steps-and-workflows/bitrise.yml
+| Environment Variable | Description |
+| --- | --- |
+| BITRISE_GO_PACKAGES | List of go packages |
 
-7. Once you're done just commit your changes & create a Pull Request
+## Contributing
```

Comment:
In this section, I think we could give a bit more guidance for successful contribution. 

1. If you want to change anything on the step you want to test it locally (and for this, in an early stage you do not want to run the step's whole test workflow). You can use a temporary workflow like this:

```
  my-workflow:
    steps:
    - path::./:
        inputs: 
        - exclude: "my exclude pattern"
```

(Do not commit the temporary workflow.)

2. Before opening a Pull Request make sure the `test` workflow passes (the step's CI build runs this workflow and it needs to pass before we can merge any changes).

Reply:
shouldn't we move this to the devcenter how to contribute guide and we just link to it? would be easier if we need to update

Reply:
Agree, Devcenter would be a better place for this information, I think we should extend the `For Contributors` section: https://devcenter.bitrise.io/contributors/contributors-index/